hi-ml-histopathology/src/histopathology/preprocessing/create_panda_tiles_dataset.py

```diff
@@ -60,109 +79,114 @@ def save_image(array_chw: np.ndarray, path: Path) -> PIL.Image:
 
 
 def generate_tiles(sample: dict, tile_size: int, occupancy_threshold: float) \
-        -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, int]:
+        -> Tuple[np.ndarray, np.ndarray, List[Box], np.ndarray, int]:
     image_tiles, tile_locations = tiling.tile_array_2d(sample['image'], tile_size=tile_size,
                                                        constant_values=255)
+    assert tile_locations.ndim == 2
     mask_tiles, _ = tiling.tile_array_2d(sample['mask'], tile_size=tile_size, constant_values=0)
 
     selected: np.ndarray
     occupancies: np.ndarray
     selected, occupancies = select_tile(mask_tiles, occupancy_threshold)  # type: ignore
-    n_discarded = (~selected).sum()
-    logging.info(f"Percentage tiles discarded: {round(selected.sum() / n_discarded * 100, 2)}")
+    num_selected = selected.sum()
+    num_tiles = len(image_tiles)
+    num_discarded = num_tiles - num_selected
+    percentage_discarded = 100 * num_discarded / num_tiles
+    logging.info(f"Discarded {num_discarded}/{num_tiles} tiles ({percentage_discarded:.2f} %)")
 
     image_tiles = image_tiles[selected]
     mask_tiles = mask_tiles[selected]
     tile_locations = tile_locations[selected]
     occupancies = occupancies[selected]
 
-    abs_tile_locations = (sample['scale'] * tile_locations + sample['location']).astype(int)
+    top, left = sample['location']
+    offset = np.array((left, top))
+
+    abs_tile_locations = (sample['scale'] * tile_locations + offset).astype(int)
+    tile_size_scaled = int(tile_size * sample['scale'])
+    tile_boxes = [Box(x, y, tile_size_scaled, tile_size_scaled) for x, y in abs_tile_locations]
 
-    return image_tiles, mask_tiles, abs_tile_locations, occupancies, n_discarded
+    return image_tiles, mask_tiles, tile_boxes, occupancies, num_discarded
```

Comment:
shall we test this function? it contains a lot of logic around offset/scale

Reply:
The idea of this PR was to write a quick fix for PANDAS. It turned out to be not that quick due to the bugs. A similar function is used in the more generic script `create_tiles_dataset`. Discussing with @dccastro, we thought we could revisit and merge these scripts if we ever needed to generate more tiles datasets, which might be unlikely as we're moving towards online cropping. Then, we would move this logic outside of the script and write tests to cover it. Does that make sense?

Reply:
I would still be in favor of writing a test just for this - if something is still wrong in the logic there will be errors in the recreated tiles. I am OK to wait for tests that concerns future datasets

Reply:
I've added a test.